components/gitpod-db/src/typeorm/migration/1657702361007-AddWebhookEvent.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class AddWebhookEvent1657702361007 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(
+            "CREATE TABLE IF NOT EXISTS `d_b_webhook_event`(`id` char(36) NOT NULL, `creationTime` varchar(255) NOT NULL, `type` char(255) NOT NULL, `status` char(255) NOT NULL, `message` text NULL, `rawEvent` text NOT NULL, `cloneUrl` char(255) NULL, `projectId` char(36) NULL, `prebuildId` char(36) NULL, `deleted` tinyint(4) NOT NULL DEFAULT '0', `_lastModified` timestamp(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6), PRIMARY KEY(`id`), KEY `ind_dbsync` (`_lastModified`), KEY `ind_cloneUrl` (`cloneUrl`), KEY `ind_creationTime` (`creationTime`), KEY `ind_projectId` (`projectId`)) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4;",
```

Comment:
type, creationTime and status are hot candidates for an index and seem very big (255 chars): For those it makes sense to make them a tad short if possible (in the past we did 60chars?), because that directly affects index size.

Also, `message` is in here, but not the `DBWebhookEvent`: I think we should have it to explain details to users, what do you think?

Reply:
> Also, message is in here, but not the DBWebhookEvent: I think we should have it to explain details to users, what do you think?

Going to add it. 

@geropl, isn't an index on `creationTime` needed to speed up "fetch last N entries"?

Reply:
> @geropl, isn't an index on creationTime needed to speed up "fetch last N entries"?

Jep, true! :bulb: